pkg/controller/jobframework/validation.go

```diff
@@ -78,7 +77,6 @@ func ValidateJobOnUpdate(oldJob, newJob GenericJob, defaultQueueExist func(strin
 	allErrs := validateUpdateForQueueName(oldJob, newJob, defaultQueueExist)
 	allErrs = append(allErrs, validateUpdateForPrebuiltWorkload(oldJob, newJob)...)
 	allErrs = append(allErrs, validateUpdateForMaxExecTime(oldJob, newJob)...)
-	allErrs = append(allErrs, validateJobUpdateForWorkloadPriorityClassName(oldJob, newJob)...)
```

Comment:
Does the PR support mutating for MultiKueue? If yes, then please test, if not, then please add validation that when multikueue is used it still cannot be mutated. 

Eventually we want to support MultiKueue, but I imagine it can be tricky to update the running Job on the worker cluster as it requires propagation as:
job -> workload -> mirrorWorkload -> mirrorJob(?)

We can work on MultiKueue in a follow up.